tests/parser/test_jsonschema.py

```diff
@@ -142,6 +148,62 @@ def test_parse_object(source_obj, generated_classes):
     ],
 )
 def test_parse_any_root_object(source_obj, generated_classes):
-    parser = JsonSchemaParser(BaseModel, CustomRootType)
+    parser = JsonSchemaParser(
+        BaseModel, CustomRootType, data_model_field_type=DataModelField
+    )
     parser.parse_root_type('AnyObject', JsonSchemaObject.parse_obj(source_obj))
     assert dump_templates(list(parser.results)) == generated_classes
+
+
+@pytest.mark.parametrize(
+    'source_obj,generated_classes',
+    [
+        (
+            {
+                "properties": {
+                    "item": {
+                        "properties": {
+                            "timeout": {
+                                "oneOf": [{"type": "string"}, {"type": "integer"}]
+                            }
+                        },
+                        "type": "object",
+                    }
+                }
+            },
+            """class Item(BaseModel):
+    timeout: Optional[Union[str, int]] = None
+
+    @validator(timeout)
+    def validate_timeout(self, value: Any) -> Any:
+        fields: List = [
+            ModelField(
+                name=str(type_), type_=type_, class_validators=None, model_config=BaseConfig
+            )
+            for type_ in [str,int]
+        ]
+        valid_item: Any = None
+        match: bool = False
+        for field in fields:  # type: ModelField
+            value, errors = field.validate(value, values={}, loc=(), cls=None)
+            if not errors:
+                if valid_item:
+                    raise ValueError(f'oneOf accept only one type. {value}')
+                valid_item = value
+                match = True
+        if match:
+            return valid_item
+        raise ValueError(f'oneOf accept only one type. {value}')
```

Comment:
@FlorianLudwig 
How do you think this validation logic for `oneOf`?

Reply:
I couldn't get it working but I think the logic wont work.  If both validators are run after each other and it is an error if both are okay this will lead to false positives.  As pydantic as a parsing and not a validating library it does accept "1" as an integer. But also as an string. So it will raise an error.

Just having `Union[int, str]` without an extra validator and letting pydantic do its thing works fine for my use case.

Reply:
OK, I remove the validator. `Union` may not cover all use-case. but, It's good start point.